### ISSUE TITLE
Offhand reinforcement

### ISSUE THREAD

**Issue by ProgrammerDan** [Civcraft/Citadel#183]
If you have _only_ reinforcement materials in the offhand and none in inventory, fortification mode exits on next block placement attempt.

Similarly since offhand is consumed _first_ vs. expectation (inventory consumed first) on both fortify and reinforce, the advantages of auto-magic reinforce are minimized.

Both /ctf and /ctr should consume from main inventory first and offhand last.

**Comment by Maxopoly** [comment]
The issue here is that /ctr and /ctf and the newfriend commandfree reinforcing call the same reinforcement method. I made it so it consumes offhand first (previously it was completly ignored) with the feature in mind where they can just put the reinforcement material in offhand and place, if the material wouldnt get consumed from offhand there, it might be weird. Not sure whats the best solution here.

**Comment by ProgrammerDan** [comment]
Few notes:
* Automatic fortification only activates if reinforcement material is in offhand _and_ in inventory.
* If INFO mode is on, auto-fortify won't activate. 
* As stated, offhand is consumed first, minimizing effectiveness.
* For stone reinforcing stone (or any block reinforcing the same type) offhand is consumed 2x and block being placed is consumed, so 3 blocks are consumed to place a single reinforced block.
* The above is true both for formal CTF mode and for auto-CTF. 

These are for myself; I'll be addressing these issues shortly.

**Comment by rourke750** [comment]
Sounds like spigot fixed the bug where it would not consume enough. Obly
took them 2 years....

**Comment by rourke750** [comment]
Is that only with off hand though? What happens if you ctf and place with
primary hand.

**Comment by ProgrammerDan** [comment]
Only offhand, so it's some new special logic. Yay!

**Comment by Maxopoly** [comment]
The normal material consumption code would only iterate over the inventory (which doesn't include offhand, armour slots and crafting inventory), so I added special handling for the off hand slot, but seems like I messed something up for forgot to change something

**Comment by ProgrammerDan** [comment]
Yeah, not a problem. I missed it for two weeks myself :D I'd always rather over-consume then under (yay dupes!) so I'm not as worried here.
